test/unit/lib/plugins/aws/deploy-function.test.js

```diff
@@ -304,6 +306,327 @@ describe('AwsDeployFunction', () => {
       });
     });
   });
+
+  describe('#mergeEnvironmentVariablesWithRemove', () => {
```

Comment:
Let's not introduce tests that are not run through `runServerless` (ideally we should not test implementation details but purely the outcome, as otherwise refactors can turn very painful - we've once also grouped this requirement here: https://github.com/serverless/serverless/tree/main/test#unit-tests)